gsa/src/web/entities/__tests__/bulkactions.js

```diff
@@ -0,0 +1,439 @@
+
+    await wait();
+
+    expect(gmp.tags.getAll).toHaveBeenCalled();
+
+    const tag = getByTestId('select-selected-value');
+    expect(tag).not.toHaveAttribute('title');
+
+    const newIcon = getByTestId('svg-icon');
+    expect(newIcon).toHaveAttribute('title', 'Create a new Tag');
+
+    fireEvent.click(newIcon);
+
+    await wait();
+
+    expect(baseElement).toHaveTextContent('New Tag');
+
+    const formFields = baseElement.querySelectorAll('input');
+
+    const selectedElements = getAllByTestId('select-selected-value');
+    expect(selectedElements.length).toEqual(2);
+    expect(selectedElements[1]).toHaveAttribute('title', 'Task');
+
+    fireEvent.change(formFields[0], {target: {value: 'goat'}});
+    expect(formFields[0]).toHaveAttribute('value', 'goat');
+
+    const saveButtons = getAllByTestId('dialog-save-button');
+    fireEvent.click(saveButtons[1]);
+
+    await wait();
+
+    expect(gmp.tag.create).toHaveBeenCalled();
+    expect(renewSessionResultFunc).toHaveBeenCalled();
+
+    await wait();
+
+    const selectedTag = getByTestId('select-selected-value');
+    expect(selectedTag).toHaveAttribute('title', 'goat');
+  });
+});
+
+const dummyEntities = [{id: 'foo'}, {id: 'bar'}];
+const selectedEntities = [{id: 'foo'}, {id: 'bar'}, {id: 'lorem'}];
+const onDownload = jest.fn();
+const exportByFilterFunc = jest.fn().mockResolvedValue({
+  data: {
+    exportIpsumByFilter: {
+      exportedEntities: '<get_entities_response />',
+    },
+  },
+});
+const exportByIdsFunc = jest.fn().mockResolvedValue({
+  data: {
+    exportIpsumByIds: {
+      exportedEntities: '<get_entities_response />',
+    },
+  },
+});
+
+const BulkExportEntitiesComponent = ({selectionType}) => {
+  const [message, setMessage] = useState('Not called');
+
+  const bulkExportEntities = useBulkExportEntities();
+
+  const handleExportEntities = () => {
+    return bulkExportEntities({
+      entities: dummyEntities,
+      selected: selectedEntities,
+      filter: Filter.fromString('cat'),
+      resourceType: 'ipsum',
+      selectionType,
+      exportByFilterFunc,
+      exportByIdsFunc,
+      onDownload,
+      onError: jest.fn(),
+    });
+  };
+
+  return (
+    <div>
+      <button
+        data-testid="load"
+        onClick={() =>
+          handleExportEntities().then(setMessage('Bulk export called!'))
+        }
+      />
+      <span data-testid="message">{message}</span>
+    </div>
+  );
+};
+
+describe('useBulkExportEntities tests', () => {
+  test('should call export by ids query for export by page content', async () => {
+    const {render} = rendererWith({store: true});
+
+    const {getByTestId} = render(
+      <BulkExportEntitiesComponent selectionType="0" />,
+    );
+
+    const message = getByTestId('message');
+
+    expect(message).toHaveTextContent('Not called');
+
+    const button = screen.getByTestId('load');
+    fireEvent.click(button);
+
+    await wait();
+
+    expect(exportByIdsFunc).toHaveBeenCalled();
```

Comment:
```suggestion
    expect(exportByIdsFunc).toHaveBeenCalled();
    expect(exportByFilterFunc).not.toHaveBeenCalled();
```
I would always check both functions for the sake of completeness so we can be sure that only the correct one is called.